app/models/spree/payment.rb

```diff
@@ -107,22 +107,19 @@ def build_source
       source.user_id = order.user_id if order
     end
 
+    # Pin payments lacks void and credit methods, but it does have refund
+    # Here we swap credit out for refund and remove void as a possible action
```

Comment:
Looking at this now I see this could be done in pin.rb instead of here. Not in scope for this PR.